crypto/provider_conf.c

```diff
@@ -335,10 +335,11 @@ static int provider_conf_load(OSSL_LIB_CTX *libctx, const char *name,
         }
         if (ok)
             ok = provider_conf_params(NULL, &entry, NULL, value, cnf);
-        if (ok && (entry.path != NULL || entry.parameters != NULL))
+        if (ok == 1 && (entry.path != NULL || entry.parameters != NULL)) {
             ok = ossl_provider_info_add_to_store(libctx, &entry);
-        if (!ok || (entry.path == NULL && entry.parameters == NULL)) {
-            ossl_provider_info_clear(&entry);
+            if (!ok || (entry.path == NULL && entry.parameters == NULL)) {
+                ossl_provider_info_clear(&entry);
+            }
```

Comment:
This does not seem to be right - what if `entry.path == NULL && entry.parameters == NULL`? The entry will be leaked. You shouldn't move this inside the previous if()

Reply:
It is also indicated by the failed CI